pallets/market/src/lib.rs

```diff
@@ -1112,6 +1206,9 @@ pub mod pallet {
                     return Err(e.into());
                 }
 
+                // Safety check on deal parameters, these should have been checked before publishing
```

Comment:
Nit: I'm afraid the "should have been" may lead someone to mistakenly think this line isn't necessary 
```suggestion
                // Safety check on deal parameters, these should be checked by the submitter before publishing
```